lib/url-parse.js

```diff
@@ -14,36 +13,38 @@ var url = require('url')
 percentEncode = function percentEncode(c) {
   var hex = c.toString(16).toUpperCase();
   (hex.length === 1) && (hex = '0' + hex)
-  return '%' + hex
+  return PERCENT + hex
 }
 
 
 /**
- * Percent encodes a query string according to RFC 3986
+ * Percent partialEncode a query string according to RFC 3986
  *
  * @param value
  * @returns {string}
  */
-encode = function (value) {
+partialEncode = function (value) {
   if (!value) { return '' }
 
   var buffer = new Buffer(value),
     ret = '',
     i
 
   for (i = 0; i < buffer.length; ++i) {
-    if (buffer[i] < 0x21 || buffer[i] > 0x7E || buffer[i] === 0x22 || buffer[i] === 0x23 ||
```

Comment:
The only issue is that if we encode `%` then we risk double encoding an already encoded URL. We should ideally encode % only if it is not followed by a hex. @czardoz?